src/Microsoft.OData.Core/UriParser/Visitors/PathSegmentToResourcePathTranslator.cs

```diff
@@ -176,6 +176,17 @@ public override string Translate(CountSegment segment)
             return "/" + segment.Identifier;
         }
 
+        /// <summary>
+        /// Translate a FilterSegment
+        /// </summary>
+        /// <param name="segment">the segment to Translate</param>
+        /// <returns>Defined by the implementer.</returns>
+        public override string Translate(FilterSegment segment)
+        {
+            Debug.Assert(segment != null, "segment != null");
+            return "/" + segment.Identifier;

```

Comment:
Is this correct?

Reply:
why do you mention to modifi to return "$filter"? It's a referenceSegment.

---
In reply to: [170140762](https://github.com/OData/odata.net/pull/1069#discussion_r170140762) [](ancestors = 170140762)